api/config/v1alpha1/configuration_types.go

```diff
@@ -83,6 +83,12 @@ type ControllerMetrics struct {
 	// It can be set to "0" to disable the metrics serving.
 	// +optional
 	BindAddress string `json:"bindAddress,omitempty"`
+
+	// CertDir is the directory that contains the server key and certificate.
```

Comment:
For kueue, we weren't really sure if there was any need to have this Cert configurable.

Any reason why we can't just default to a hardcoded location?

Reply:
If we mount the default hardcoded locations `/tmp/k8s-metrics-server/serving-certs`, `tls.crt`, `tls.key`, there won't be any code changes needed in main.go.

Reply:
I think there would be one minor change in setting the right tlsOpts for metrics server. No having this CertDir does simply the code.

But if there is a reason for it I think leaving it in is fine.

Reply:
In kustomize and helm we offer prometheus support without cert manager. This gives the users a way to inject their certificates for the metrics server to the deployment.

We also offer the same thing for the webhook.